Mathlib/NumberTheory/ArithmeticFunction.lean

```diff
@@ -1073,12 +1073,17 @@ theorem moebius_ne_zero_iff_squarefree {n : ℕ} : μ n ≠ 0 ↔ Squarefree n :
   · simp [h, pow_ne_zero]
 #align nat.arithmetic_function.moebius_ne_zero_iff_squarefree ArithmeticFunction.moebius_ne_zero_iff_squarefree
 
+theorem moebius_eq_or (n : ℕ) : μ n = 0 ∨ μ n = 1 ∨ μ n = -1 := by
```

Comment:
To me this looks like an argument to change the return type to be `SignType`

Reply:
To push this further, I guess one could argue that `ArithmeticFunction.zeta` should take values in `Fin 2`...

Reply:
I'm suspect that the coercions aren't really a problem; and in exchange, you get results like this and the `abs` one for free

Reply:
(if the coercions were really that bad, then presumably `SignType` wouldn't exist at all!)

Reply:
Could you link to that review?

Reply:
#10883. See also #10920, which was some API additions for `SignType` which I had to add to get #10883 working.

Reply:
FWIW, I just had a stab at reimplementing `μ` to be `SignType`-valued, just to check how bad it would be. It's every bit as bad as I expected. Literally every single lemma in the library about `μ` would need to be rewritten, with clunkier statements and longer proofs. We'd also need to add a whole load of tedious boilerplate code about coercions `ArithmeticFunction SignType → ArithmeticFunction R` for appropriate `R`, and compatibility of these with morphisms between `R`'s. This is really, really not the way to go.

Reply:
Thanks for the cross-links. I'll have a think about this some more later, but certainly actually changing this isn't a blocker for this PR.

Reply:
> FWIW, I just had a stab at reimplementing `μ` to be `SignType`-valued, just to check how bad it would be

Would you mind sharing the branch with this stab on? No need to PR it, but if you put in the work it's nice to refer back to if this comes up again in future.

Reply:
I deleted it, sorry.